pkg/clusterversion/clusterversion.go

```diff
@@ -90,3 +98,248 @@ func (builder *Builder) validate() (bool, error) {
 
 	return true, nil
 }
+
+// WithDesiredUpdateImage adds the desired image to the clusterversion struct.
+func (builder *Builder) WithDesiredUpdateImage(desiredUpdateImage string, force bool) (*Builder, error) {
+	var err error
+
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	if desiredUpdateImage == "" {
+		glog.V(100).Infof("The desiredUpdateImage is empty")
+
+		return builder, fmt.Errorf("'desiredUpdateImage' cannot be empty")
+	}
+
+	glog.V(100).Info("Updating clusterversion %s to the desired image %s",
+		builder.Definition.Name, desiredUpdateImage)
+
+	builder.Definition.Spec.DesiredUpdate = &v1.Update{Image: desiredUpdateImage, Force: force}
+
+	return builder, err
+}
+
+// WithDesiredUpdateChannel adds the desired channel to the clusterversion struct.
+func (builder *Builder) WithDesiredUpdateChannel(updateChannel string) (*Builder, error) {
+	var err error
```

Comment:
I think you can safely remove this line

Reply:
same comment as above

Reply:
Yes because With* functions can return builder only. Please remove error from return and use builder.errorMsg instead. Please refer to:

[eco-goinfra/pkg/sriov/network.go](https://github.com/openshift-kni/eco-goinfra/blob/4b536d23d57aedc397057a32d67fd2823296326e/pkg/sriov/network.go#L105)

Line 105 in [4b536d2](https://github.com/openshift-kni/eco-goinfra/commit/4b536d23d57aedc397057a32d67fd2823296326e)

 builder.errorMsg = "invalid 'vlanProtocol' parameters"

Reply:
Removed, thanks